torch/utils/_config_module.py

```diff
@@ -157,7 +161,13 @@ def __delattr__(self, name: str) -> None:
         self._is_dirty = True
         # must support delete because unittest.mock.patch deletes
         # then recreate things
-        del self._config[name]
+        self._config[name].user_override = None
+
+    def _is_default(self, name: str) -> bool:
+        return (
+            self._config[name].user_override is None
+            or self._config[name].user_override == self._config[name].default
```

Comment:
This test is suspicious. I can see why you did it this way, but an alternate semantics for "is default" is to always report False if the user has done an overwrite. This is more stable if we change defaults.

Reply:
updated to just checking if the sentinel is set.

Note that with the above changes to allow reference behaviour, we'll slightly overreport is_default(key) != true for reference types, since on access, we'll mark them as having potentially mutated.